tests/fe/uneven_1d_grid_hermite.cc

```diff
@@ -0,0 +1,192 @@
+#include <deal.II/numerics/matrix_tools.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <cmath>
+#include <cstdlib>
+#include <fstream>
+#include <iterator>
+#include <sstream>
+#include <vector>
+
+
+
+/*
+ * Test case for Hermite on an irregular 1D grid.
+ * <code>FE_Hermite<1>(reg)<\code> should be able to perfectly represent any
+ * polynomial function up to degree $2 \times reg+1$. If all basis functions
+ * are correctly scaled according to element size, then projecting a
+ * polynomial of this form onto the Hermite FE space will produce negligible
+ * pointwise errors.
+ */
+using namespace dealii;
+
+// Define a function to project onto the domain
+class test_poly : public Function<1>
+{
+public:
+  virtual double
+  value(const Point<1> &p, unsigned int c = 0) const override
+  {
+    return p(c) * (1.0 + p(c) * (0.5 - p(c)));
+  }
+
+
+
+  std::string
+  get_polynomial_string()
+  {
+    return "X + 0.5 X^2 - X^3";
+  }
+};
+
+
+
+void
+test_fe_on_domain(const unsigned int regularity)
+{
+  Triangulation<1> tr;
+  DoFHandler<1>    dof(tr);
+
+  double   left = -1.0, right = 1.0;
+  Point<1> left_point(left), right_point(right);
+  GridGenerator::hyper_cube(tr, left, right);
+
+  // Refine the right-most cell three times to get the elements
+  // [-1,0],[0,0.5],[0.5,0.75],[0.75,1]
+  for (unsigned int i = 0; i < 3; ++i)
+    {
+      for (auto &cell : tr.active_cell_iterators())
+        {
+          const double distance = right_point.distance(cell->vertex(1));
+          if (distance < 1e-6)
+            {
+              cell->set_refine_flag();
+              // break;
+            }
+        }
+      tr.execute_coarsening_and_refinement();
+    }
+
+  FE_Hermite<1> herm(regularity);
+  dof.distribute_dofs(herm);
+
+  MappingCartesian<1> mapping;
+
+  QGauss<1>      quadr(2 * regularity + 2);
+  Vector<double> solution(dof.n_dofs());
+  test_poly      rhs_func;
+
+  AffineConstraints<double> constraints;
+  constraints.close();
+
+  FEValues<1> fe_herm(mapping,
+                      herm,
+                      quadr,
+                      update_values | update_quadrature_points |
+                        update_JxW_values);
+
+  std::vector<types::global_dof_index> local_to_global(herm.n_dofs_per_cell());
+
+  VectorTools::project_hermite(
+    mapping, dof, constraints, quadr, rhs_func, solution, false);
+
+  double err_sq = 0;
+
+  for (auto &cell : dof.active_cell_iterators())
+    {
+      fe_herm.reinit(cell);
+      cell->get_dof_indices(local_to_global);
+      for (const unsigned int q : fe_herm.quadrature_point_indices())
+        {
+          double sol_at_point = 0;
+          for (const unsigned int i : fe_herm.dof_indices())
+            sol_at_point +=
+              fe_herm.shape_value(i, q) * solution(local_to_global[i]);
+
+          sol_at_point -= rhs_func.value(fe_herm.quadrature_point(q));
+          err_sq += sol_at_point * sol_at_point * fe_herm.JxW(q);
+        }
+    }
```

Comment:
Do yo need to compute this by hand? Cannot you use functions from the `VectorTools` namespace.